components/Navigation.tsx

```diff
@@ -15,24 +22,66 @@ export type StackNavigatorType<
   ParamsList extends Record<string, object | undefined>
 > = ReturnType<typeof createStackNavigator<ParamsList>>
 
+/**
+ * Returns an object of functions to navigate to essential app screens.
+ */
+export const useCoreNavigation = () => {
+  // TODO: - Actually Navigate
+  const navigation = useNavigation()
+  return {
+    presentEditEvent: (edit: EditEventFormValues, id?: EventID) => {
+      const formValues = toRouteableEditFormValues(edit)
+      if (id) {
+        navigation.navigate("editEvent", {
+          screen: "editEventForm",
+          params: { ...formValues, id }
+        })
+      } else {
+        navigation.navigate("editEvent", {
+          screen: "createEventForm",
+          params: formValues
+        })
+      }
+    },
+    presentProfile: (
+      id: UserID | UserHandle,
+      method: "navigate" | "replace" = "navigate"
+    ) => {
+      ;(navigation as any)[method]("editEvent", {
+        screen: "userProfile",
+        params: { id, method }
+      })
+    },
+    pushEventDetails: (
+      id: EventID,
+      method: "navigate" | "replace" = "navigate"
+    ) => {
+      // NB: Typescript doesn't like the subscript for some reason, but navigation should always
+      // have a navigate and replace method.
+      ;(navigation as any)[method]("eventDetails", { id, method })
+    },
```

Comment:
Why does the syntax have a ";" before the use of `(navigation as any)`? Is that Prettier doing that because of a weird way the functions are formatted?

Reply:
Yes...